classes/Mail.php

```diff
@@ -293,9 +297,22 @@ function ($carry, $item) {
             return false;
         }
 
-        /* Construct multiple recipients list if needed */
-        $message = new Swift_Message();
+        /* Create new message and DKIM sign it, if enabled and all data for signature are provided */
+        if ($configuration['PS_MAIL_DKIM_ENABLE'] == 1 && !empty($configuration['PS_MAIL_DKIM_DOMAIN']) && 
+            !empty($configuration['PS_MAIL_DKIM_SELECTOR']) && !empty($configuration['PS_MAIL_DKIM_KEY'])) {
+            
+            $message = new Swift_SignedMessage();
+            $privateKey = $configuration['PS_MAIL_DKIM_KEY'];
+            $domainName = $configuration['PS_MAIL_DKIM_DOMAIN'];
+            $selector = $configuration['PS_MAIL_DKIM_SELECTOR'];
+            $signer = new Swift_Signers_DKIMSigner($privateKey, $domainName, $selector);
```

Comment:
```suggestion
            $signer = new Swift_Signers_DKIMSigner(
                $configuration['PS_MAIL_DKIM_KEY'],
                $configuration['PS_MAIL_DKIM_DOMAIN'],
                $configuration['PS_MAIL_DKIM_SELECTOR']
            );            
```

Array keys already have a good name for a good understanding